nodejs/week3/homework-template/src/backend/api/meals.js

```diff
@@ -0,0 +1,85 @@
+const express = require("express");
+require("express-async-errors");
+const router = express.Router();
+const knex = require("../database");
+const { sum, limit } = require("../database");
+
+//to inser a new meal
+router.post("/", async (req, res) => {
+	const newMeal = {
```

Comment:
what happens if the request provides wrong data? e.g. the price is not a number?

A good advice is to never trust data provided by a user (or a client/consumer) of an API or application. You need to validate any type of data that you receive from the outside world so that it doesn't break your application in unexpected ways. If an error occurs you want to communicate to the client of your application what the error is and how they can recover from that error. That helps you troubleshoot issues and guides your development.

(e.g. if I get a 404 not found I know that the id that I'm sending is probably wrong, if I get a 400 bad request I know that there's some parameter in my request that is wrong, etc)

Reply:
I used global error handler, express-async-errors package. Would it  be right?